packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js

```diff
@@ -1,10 +1,46 @@
 import {LightningElement, api} from 'lwc';
+import LOCALE from '@salesforce/i18n/locale';
 
 export default class QuanticFacetValue extends LightningElement {
-  /** @type {import("coveo").FacetValue} */
   @api item;
   /** @type {boolean} */
   @api isChecked;
+  /** @type {string} */
+  @api variant;
+  /** @type {string} */
+  @api start;
+  /** @type {string} */
+  @api end;
+
+  Variants = {
```

Comment:
I would say we want to have a default facet value name, example `StandardFacet` .
We can do 
`@api variant = 'StandardFacet';`

And maybe even create multiple templates like this?

https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.create_render

Reply:
I think either we have different templates per variant.

Or we have ONE template, but you make a method called:

```js
get formattedFacetValue() {
  if(this.isDateFacet) {
    return `${this.start} - ${this.end}`;
  } else if (this.isNumericFacet) {
    return [...]
  }
  [...]
}
```

I really like the idea of using a variant attribute.
I thought we could also support a `custom` variant, with another parameter that would be just a 
`@api formattingFunction;`

So you would do a `return this.formattingFunction(this.item);` ?